src/overlay_01_02204ED8.c

```diff
@@ -0,0 +1,218 @@
+static s8 ov01_02205074(u8, u8, u32);
+static void ov01_022050F8(FieldSystem*, u32*, u32*);
+
+const s8 ov01_02209634[] = {
+    0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0,
+};
+
+const s8 ov01_02209640[][2] = {
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+    {0, 1},
+    {1, 0},
+};
+
+const s8 ov01_02209654[][3] = {
+    { 0,  1, -1},
+    { 0, -1,  1},
+    { 1,  0, -1},
+    { 1, -1,  0},
+    {-1,  0,  1},
+    {-1,  1,  0},
+    { 0,  1, -1},
+    { 0, -1,  1},
+    { 1,  0, -1},
+    { 1, -1,  0},
+};
+
+const s8 ov01_02209672[][4] = {
+    {-1,  0,  0,  1},
+    { 1, -1,  0,  0},
+    { 1,  1, -1,  0},
+    { 0,  1,  1, -1},
+    { 0,  0,  1,  1},
+    {-1,  0,  1,  0},
+    { 1, -1,  0,  1},
+    { 0,  1, -1,  0},
+    { 1,  0,  1, -1},
+    { 0,  1,  0,  1},
+};
+
+const s8 ov01_0220969A[][5] = {
+    {-1,  0,  0,  1,  1},
+    { 1, -1,  0,  0,  1},
+    { 1,  1, -1,  0,  0},
+    { 0,  1,  1, -1,  0},
+    { 0,  0,  1,  1, -1},
+    {-1,  0,  1,  0,  1},
+    { 1, -1,  0,  1,  0},
+    { 0,  1, -1,  0,  1},
+    { 1,  0,  1, -1,  0},
+    { 0,  1,  0,  1, -1},
+};
+
+void ov01_02204ED8(FieldSystem *fsys, u16 *a1) {
+    u16 **r2 = AllocFromHeapAtEnd(HEAP_ID_FIELD, sizeof(r2));
+    *r2 = a1;
+    *a1 = FALSE;
+    TaskManager_Call(fsys->taskman, ov01_02204EFC, r2);
+}
+
+static BOOL ov01_02204EFC(TaskManager *taskManager) {
+    UnkStruct_02204EFC *r4;
+    FieldSystem *fsys = TaskManager_GetFieldSystem(taskManager);
+    u16 **r6 = TaskManager_GetEnv(taskManager);
+    r4 = AllocAtEndAndReadWholeNarcMemberByIdPair(NARC_a_2_5_2, fsys->location->mapId, HEAP_ID_FIELD);
+    if (r4->unk00 != 0 || r4->unk02 != 0) {
+        BATTLE_SETUP *sp10;
+        u32 x;
+        u32 y;
+        ov01_022050F8(fsys, &x, &y);
+        u32 trainerId = PlayerProfile_GetTrainerID(Save_PlayerData_GetProfileAddr(fsys->savedata));
+        s32 r0 = ov01_02204FE0(r4->unk00, r4->unk02, trainerId, x, y, &r4->unk4c);
+        if (r0 == -1) {
+            FreeToHeap(r4);
+            FreeToHeap(r6);
+            return TRUE;
+        }
+        void *r2;
+        if (r0 == 0) {
+            r2 = &r4->unk04;
+        } else if (r0 == 1) {
+            r2 = &r4->unk1c;
+        } else if (r0 == 2) {
+            r2 = &r4->unk34;
+        } else {
+            GF_ASSERT(FALSE);
+            FreeToHeap(r4);
+            FreeToHeap(r6);
+            return TRUE;
+        }
+        if (ov02_02247374(fsys, &sp10, r2)) {
+            **r6 = TRUE;
+            FreeToHeap(r4);
+            FreeToHeap(r6);
+            sub_02050B90(fsys, taskManager, sp10);
+            return FALSE;
+        }
+    }
+    FreeToHeap(r4);
+    FreeToHeap(r6);
+    return TRUE;  
+}
+
+static s8 ov01_02204FE0(u16 a0, u16 a1, u32 trainerId, u32 x, u32 y, void *a5) {
+    u16 r1 = a0 * 6;
+    u16 ip = a1 * 6;
+    for (u16 r0 = 0; r0 < r1; r0++) {
+        if (x == *(s16*)(a5 + r0 * 4)) {
+            if (y == *(s16*)(a5 + r0 * 4 + 2)) {
```

Comment:
note that here implies that a5 is a 2D array (s16 arr[c][2]) where [c][0] is an x value and [c][1] is a y value